### ISSUE TITLE
Bad icon quality

### ISSUE THREAD

**Issue by tobigit** [fontello/fontello#359]
Like some months ago we have bad icon quality in the latest Firefox (v33.1 / Window 7).
In Google Chrome the icons look good!

We tested the old fontello-font-icons from April this year with the latest Firefox (v33.1). The icons look great!

Is _"hinting"_ the causer again?

In April 2014 there was the same Problem: https://github.com/fontello/fontello/issues/276

**Comment by puzrin** [comment]
Still a problem?

**Comment by knalli** [comment]
@puzrin Yes, it is still a problem for the specific target group running a Firefox on Windows with a special GPU constellation.

See also https://bugzilla.mozilla.org/show_bug.cgi?id=1096934

**Comment by puzrin** [comment]
So, can i close this issue?

**Comment by knalli** [comment]
Do you have an idea or hint how to solve this? It is only a assumption it's caused by Firefox (my current knowledge: this is a common thing). After all it results into a bad UI.

**Comment by puzrin** [comment]
I could misunderstand something. Could you help to narrow down condition?

- what changes if you enable/disable hinting?
- what is difference between FF / Chrome?

Aslo FF 33 is no longer supported, probably need to use lstest or latest ERS version.

**Comment by puzrin** [comment]
There is a bug related to ttfautohint https://github.com/fontello/fontello/issues/345

Problem is, that 0.97 can crash with high memory/cpu consumption. And it's not clear what exactly could cause error on 0.97 -> 1.1 change.

I can try to rebuild ttfautohint to latest 1.5, may be it helps (and add another problems :)).

**Comment by knalli** [comment]
> what changes if you enable/disable hinting?

I don't know this, but @tobigit will have a look later.

> what is difference between FF / Chrome?

In Chrome is no issue (or we don't aware of).

> Aslo FF 33 is no longer supported, probably need to use lstest or latest ESR version.

Well, the issue comes present starting with Firefox 33. It is not resolved using any version greater than this one.

The ttfsautohint issue could be matching this.

**Comment by puzrin** [comment]
As soon, as we know easy to reproduce combination of (Win version + FF + hardware + TTF sample), it's possible to forward report to ttfautohint author directly and he could triage.

**Comment by puzrin** [comment]
Looks like bug can be in different components in the same time. Here is suggestion how to check ttfautohint:

1. Dowload versions from 0.97 and up. http://www.freetype.org/ttfautohint/#download
2. Install ttf2woff https://github.com/fontello/ttf2woff
2. Create font without hinting in fontello.
3. For each ttfautohint version:
    - apply ttfautohint to TTF https://github.com/fontello/fontello/blob/5.2.0/server/font/_lib/worker.js#L206-L210
    - apply ttf2woff to resulting TTF (or, may be just delete it to force TTF use)
    - check result in browser

That can help to understand, if there is version of ttfautohint with workaround of FF problem.

**Comment by puzrin** [comment]
@knalli , @tobigit could you rebuild font and check again? Probably, this bug could be related to https://github.com/fontello/fontello/issues/464#issuecomment-202244651

**Comment by knalli** [comment]
Sorry for the delay. I have personally not figured out how to create a test/verification environment, except my coworker's one but he has not got into into it yet. 

What do you mean with rebuild? Against the published version online, or a specific one at the repo?

**Comment by puzrin** [comment]
I meaned - import your config to fontello (with enabled hinting), download font and check if bug still exists or not with it.

**Comment by tobigit** [comment]
The new generated font icons have the same bad quality. They looks a bit different but not as good as the version of April 2014.

In Chrome they look great:
![font-icons-chrome](https://cloud.githubusercontent.com/assets/1849231/14247169/91923846-fa6c-11e5-9ffc-bdf117e9622e.PNG)

In Firefox...
1: Our actual Version
2: New generated today with "hinting"
3: New generated today without "hinting"
![font-icons-firefox](https://cloud.githubusercontent.com/assets/1849231/14247174/95759fa2-fa6c-11e5-96b2-7c82b2b098a0.png)

**Comment by puzrin** [comment]
I see icons are different, but can't say "everything better" or "everything worse".

On the middle example top right icon is worse, bottom right is better.

**Comment by knalli** [comment]
@puzrin Yes, but it not like Chrome or Firefox in regular. The first one is much better than the three beneath.

**Comment by puzrin** [comment]
@knalli there are no doubts that crome sample is better, but we can do nothing with this.

As far as i understand, question is in difference between 3 last samples. Initial posts say, it was good, and become worse. With last 3 images i don't see significant change (something become worse, something become better, and general difference is not huge).

Do i miss something?
